### PR TITLE
Added sources argument to the news command

### PR DESCRIPTION
**Pull request by Flodur871**
I added an argument that allows filtering of news by source, as was mentioned in the ```#TODO```.

### CODE REVIEW

**Comment by DidierRLopes** [comment]
I think this error will be fixed when #754 is merged

File: gamestonk_terminal/stocks/stocks_controller.py [hunk 1]
```diff
@@ -251,6 +251,14 @@ def call_news(self, other_args: List[str]):
             default=True,
             help="Show oldest articles first",
         )
+        parser.add_argument(
+            "-s",
+            "--sources",
+            dest="n_sources",
+            default=[],
```
Comment (jmaslek) [inline gamestonk_terminal/stocks/stocks_controller.py:258]: So the "n_" in dest (if used) is used for number variables (int or floats).  So can we can either name this "s_sources", (s for string) or just "sources".

File: gamestonk_terminal/common/newsapi_view.py [hunk 2]
```diff
@@ -24,14 +25,19 @@ def news(
...
-    response = requests.get(
-        f"https://newsapi.org/v2/everything?q={term}&from={s_from}"
-        f"&sortBy=publishedAt&language=en&apiKey={cfg.API_NEWS_TOKEN}",
+    link = (
+        f"https://newsapi.org/v2/everything?q={term}&from={s_from}&sortBy=publishedAt&language=en"
+        f"&apiKey={cfg.API_NEWS_TOKEN}"
     )
 
+    if sources:
+        link += f"&domains={sources}"
```
Comment (jmaslek) [inline gamestonk_terminal/common/newsapi_view.py:37]: Could we check that each source ends in a .com, and if not add it, this way one could just do `news -s yahoo`

File: gamestonk_terminal/stocks/stocks_controller.py (hunk 1 above)
Comment (Flodur871) [reply gamestonk_terminal/stocks/stocks_controller.py:258]: I see

File: gamestonk_terminal/common/newsapi_view.py (hunk 2 above)
Comment (Flodur871) [reply gamestonk_terminal/common/newsapi_view.py:37]: Sure

File: gamestonk_terminal/stocks/stocks_controller.py [hunk 3]
```diff
@@ -254,23 +254,27 @@ def call_news(self, other_args: List[str]):
...
-            help="Show news only from the sources specified (e.g bbc.com yahoo.com)",
+            help="Show news only from the sources specified (e.g bbc yahoo.com)",
         )
 
         try:
             ns_parser = parse_known_args_and_warn(parser, other_args)
             if not ns_parser:
                 return
 
+            sources = ns_parser.sources
```
Comment (DidierRLopes) [inline gamestonk_terminal/stocks/stocks_controller.py:267]: A more elegant solution would be:
```
sources = 
[src for src in ns_parser.sources if ".com" in src] + [f"{src}.com" for src in ns_parser.sources if ".com" not in src]
```

list comprehensions are quite powerful :)

File: gamestonk_terminal/stocks/stocks_controller.py [hunk 4]
```diff
@@ -254,23 +254,27 @@ def call_news(self, other_args: List[str]):
         parser.add_argument(
             "-s",
             "--sources",
-            dest="n_sources",
             default=[],
             nargs="+",
-            help="Show news only from the sources specified (e.g bbc.com yahoo.com)",
+            help="Show news only from the sources specified (e.g bbc yahoo.com)",
```
Comment (DidierRLopes) [inline gamestonk_terminal/stocks/stocks_controller.py:259]: you're missing a comma here between bbc and yahoo

**Review by DidierRLopes — APPROVED** [review]
I commented 2 tiny things, but looks good to me :)

Great work!

File: gamestonk_terminal/stocks/stocks_controller.py (hunk 4 above)
Comment (jmaslek) [reply gamestonk_terminal/stocks/stocks_controller.py:259]: Doesn't "nargs=+" eliminate the need for commas?

Comment (DidierRLopes) [reply gamestonk_terminal/stocks/stocks_controller.py:259]: You're right. I saw the `",".join()` and got confused, but we just process it like that after. So all good

**Pull request merged**